connection.go

```diff
@@ -404,17 +416,27 @@ func (c *Connection) writeMessage(w io.Writer, message *iso8583.Message) error {
 		return utils.NewSafeError(&PackError{err}, "failed to pack message")
 	}
 
+	// create buffer for header and packed message so we can write it to
+	// the connection as a single write
+	buf := &bytes.Buffer{}
+
 	// create header
-	_, err = c.writeMessageLength(w, len(packed))
+	_, err = c.writeMessageLength(buf, len(packed))
```

Comment:
Are we sure this is flushing? Byte buffers don't have locks or guarantee atomic operations.

Reply:
`c.writeMessageLength` writes encoded len into `buf`. `buf` has no `Flush`